### PR TITLE
Add deprecation note to Dynatrace metrics exporter

### PR DESCRIPTION
**Pull request by joaopgrassi**
**Description:**
Dynatrace has support for ingesting traces, metrics and logs via OTLP, so the custom metrics exporter is not needed anymore. This PR adds a deprecation message about it as well as other minor wording/improvements.

**Link to tracking Issue:**
N/A

**Testing:**
N/A

**Documentation:**
Exporter readme updated.

### CODE REVIEW

File: exporter/dynatraceexporter/README.md [hunk 1]
```diff
@@ -1,24 +1,27 @@
...
-More information on exporting traces to Dynatrace can be found in the
-[Dynatrace documentation for OpenTelemetry traces](https://www.dynatrace.com/support/help/extend-dynatrace/opentelemetry/opentelemetry-traces/opentelemetry-ingest).
+[Dynatrace](https://www.dynatrace.com/integrations/opentelemetry) supports native
+OpenTelemetry protocol (OTLP) ingest for traces, metrics and logs.
+All signals can be sent directly to Dynatrace via **OTLP protobuf over HTTP**
+using the built-in OTLP/HTTP Exporter available in the collector.
+More information on using the collector with Dynatrace can be found in the
+[Dynatrace documentation](https://www.dynatrace.com/support/help/shortlink/opentelemetry-collector-explanation#configuration).
 
 # Dynatrace Exporter
```
Comment (arminru) [inline exporter/dynatraceexporter/README.md]: The first paragraph might be overlooked too easily and the extra text above the first heading looks a bit odd I think, also most markdown linters would even complain about that.

How about adding this?


```suggestion
# Dynatrace

[Dynatrace](https://www.dynatrace.com/integrations/opentelemetry) supports native
OpenTelemetry protocol (OTLP) ingest for traces, metrics and logs.
All signals can be sent directly to Dynatrace via **OTLP protobuf over HTTP**
using the built-in OTLP/HTTP Exporter available in the collector.
More information on using the collector with Dynatrace can be found in the
[Dynatrace documentation](https://www.dynatrace.com/support/help/shortlink/opentelemetry-collector-explanation#configuration).

## Dynatrace Metrics Exporter
```
(+ adding an extra # to the others headings below)

**Review by arminru — APPROVED** [review]
LGTM!

**Review by mviitane — APPROVED** [review]

**Review by codeboten — COMMENTED** [review]
Thanks @joaopgrassi! Please add a changelog notice to notify end-users of the deprecation

**Review by codeboten — APPROVED** [review]
Thanks @joaopgrassi!

**Review by codeboten — COMMENTED** [review]
One last request, please add a note in the go.mod as well, see https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/24c80cc5b5f972bd6869ab11f90095f0a026493a/exporter/jaegerexporter/go.mod#L1 for an example

**Comment by joaopgrassi** [comment]
@codeboten ah thanks for the pointer, done!

We are not ready to remove it completely, so I added the similar wording as we have in the readme for now. I hope that is OK :)

**Review by codeboten — APPROVED** [review]
Looks great!

**Pull request merged**